### ISSUE TITLE
Writing file header early when outputting to php://output

### ISSUE THREAD

**Issue by hollymcr** [PHPOffice/PhpSpreadsheet#3228]
This is:

```
- [X] a feature request
- [X] **not** a usage question
```

When creating a large spreadsheet (Xlsx for my application but I'm interested in the general case), which is to be sent to the browser via php://output with a Content-disposition header, the user has to wait until the code starts generating output before the browser will see the headers and prompt the user for a download location. If this is several minutes (as in my case) this is a usability headache.

It doesn't require much output to be flushed to the browser to fix this (subject to suitable web server and PHP configuration), indeed in my test code just the "PK" from the file header would be enough. Therefore it would be useful to be able to separate sending the file header, from sending the file content that follows, so that a script could optionally send the header early.

### What is the expected behavior?
Something like this would be great:
```
$spreadsheet = new Spreadsheet();
$writer = new Xlsx($spreadsheet);
header("Content-Disposition: attachment; filename=mytest.xlsx");
$writer->saveHeader("php://output");
// Insert code that generates the spreadsheet content here
$writer->save();
```

**Comment by oleibman** [comment]
I don't think this is anything we could do in a general case. In addition, I am unable to duplicate your results. I added the following code after the calls to the header function in 01_Simple_download_xlsx.php:
```php
echo 'PK';
flush();
sleep(60);
$outfile = 'anywhere you want to store a temporary file';

$writer = IOFactory::createWriter($spreadsheet, 'Xlsx');
$writer->save($outfile);
$contents = file_get_contents($outfile);
echo substr($contents, 2);
exit;
```
This succeeded, in a sense - the browser did download a usable spreadsheet. However, it did not prompt for the location at least until the call to `sleep` had finished, so we are no better off than when we started. If the expected behavior were as you said, I would expect the prompt to come immediately, not after 60 seconds.

So, I am closing this issue. If your environment differs from mine, and you do get the prompt early, try using the code above as a template.
